User geolocation in Google Maps

stevygee opened the pull request:
- Adds parameter 'mylocation'
- When enabled, geolocation toggle button is shown in map UI
- Adds marker and accuracy circle to map
- Watches user position, re-centers map
- When dragged, position keeps updating, but map no longer follows
- Geolocation can be turned off
- Fixes AJAX markers not updating when changing center only

Fixes #98

JeroenDeDauw commented on src/GoogleMapsService.php:
```diff
@@ -273,7 +280,7 @@ public function newMapId(): string {
 
 	public function getResourceModules( array $params ): array {
-		return [ 'ext.maps.googlemaps3', 'ext.maps.googlemaps3ajax' ];
+		return [ 'ext.maps.googlemaps3', 'ext.maps.googlemaps3ajax', 'ext.maps.mylocation' ];
```
If you always load the new js file anyway, you could put it in one of the existing modules.

stevygee replied:
Sure, I'm not very familiar how script loading works in MediaWiki, but I've added it to the `ext.maps.googlemaps3` module now.

JeroenDeDauw commented:
Do you have a demo of this somewhere? Or a short video?

Flexmaen commented:
> Do you have a demo of this somewhere? Or a short video?

Here: https://baugeschichte.at/Spezial:KU/ff9
Might mike more sense if you use it in your phone,  and with a fake location set to some point that has entries around.

stevygee commented:
Added: The user location setting is now remembered (local storage), so it remains active for all maps on a site.

Added two new parameters:
- **mylocationfollow**: When tracking user location, also follow user location by re-centering map on them (on/off).
- **mylocationzoom**: Level to zoom to when activating user location. Set at -1 to disable (default).

(both of these are disabled in the demo)

PeterTheOne commented:
Looks good to me :)
